runtime/tools/spec-builder/test/build-spec.spec.js

```diff
@@ -0,0 +1,37 @@
+const assert = require('assert');
+const fs = require('fs');
+const { buildSpec, emptyActualRawFiles } = require('./utils/spawn-child-process');
+const { isEqual } = require('lodash');
+
+
+before(async function() {
+    await emptyActualRawFiles();
+    await buildSpec('./test/json-files/customSpec.json', './test/actual-raw-files/customSpecRaw.json');
+    await buildSpec('./test/json-files/customSpec-balances.json', './test/actual-raw-files/customSpec-balances.json');
+    await buildSpec('./test/json-files/customSpec-code.json', './test/actual-raw-files/customSpecRaw-code.json');
+});
+
+describe('Tool generates raw files', function(){
+    it('successfully generates customSpec file with all properties', function(){
+        assert(fs.existsSync('./test/actual-raw-files/customSpecRaw.json'), 'file does not exist');
+    })
+    it('successfully generates customSpec file with system property only', function(){
+        assert(fs.existsSync('./test/actual-raw-files/customSpecRaw-code.json'), 'file does not exist');
+    })
+    it('should fail to convert customSpec without system property', function() {
+        assert(!fs.existsSync('./test/actual-raw-files/customSpecRaw-balances.json'), 'file exists');
```

Comment:
Normally when you have a negative test, you are checking that:
1) The operation fails (at minimum)
2) Asserting what actual error was received (if possible)
that is a second reason why I would like for you to put the `buildSpec` commands in the `it` tests. This way you will be able to expect that something reverts.

It may be the case that we need to exit the program when `system` property is not returned with `exit code 1` in order to catch the error correctly